src/nvim/eval.c

```diff
@@ -15534,6 +15534,64 @@ static void f_split(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   p_cpo = save_cpo;
 }
 
+
+/// "stdpath(type)" function
+static void f_stdpath(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  const char *p = tv_get_string(&argvars[0]);
```

Comment:
Use `tv_get_string_chk` and return early, so you would not give another error message for non-string when errorring out in `else`.